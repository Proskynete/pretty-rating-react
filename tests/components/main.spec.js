import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import PrettyRating from '../../src/main';

Enzyme.configure({ adapter: new Adapter() });

describe('<PrettyRating />', () => {
	it('should render component', () => {
		const rating = 4.5;
		const icons = {
			complete: 'fas fa-star',
			half: 'fas fa-star-half-alt',
			empty: 'far fa-star',
		};

		const component = renderer.create(
			<PrettyRating rating={rating} icons={icons} />,
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
