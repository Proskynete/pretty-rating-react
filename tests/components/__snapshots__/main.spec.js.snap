// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrettyRating /> should render component 1`] = `
Array [
  <i
    aria-hidden="true"
    className="fas fa-star"
  />,
  <i
    aria-hidden="true"
    className="fas fa-star"
  />,
  <i
    aria-hidden="true"
    className="fas fa-star"
  />,
  <i
    aria-hidden="true"
    className="fas fa-star"
  />,
  <i
    aria-hidden="true"
    className="fas fa-star-half-alt"
  />,
]
`;
